php/libraries/Module.class.inc

```diff
@@ -58,6 +58,30 @@ abstract class Module extends \LORIS\Router\PrefixRouter
         );
     }
 
+    /**
+     * Retrieve all active module descriptors from the given database, indexed
+     * by their primary key in the database.
+     *
+     * @param \Database $db an open connection to a database containing a 'modules'
+     *                      table
+     *
+     * @return \Module[]
+     */
+    static public function getActiveModulesIndexed(\Database $db): array
+    {
+        $mnames = $db->pselect(
+            "SELECT ID, Name FROM modules WHERE Active='Y'",
+            []
+        );
+
+        $rv = [];
+        foreach ($mnames as $row) {
+            $rv[$row['ID']] = self::factory($row['Name']);
+        }
```

Comment:
you can use `pselectWithIndexKey` function instead

Reply:
I didn't know that existed, but how would that work? This loop would still be required to convert the value to a `\Module` instance

Reply:
true, maybe we should add the possibility to give it a mapping function as well

Reply:
That is not within the scope of this PR

Reply:
#6021